### PR TITLE
Proxying outbound connections

### PR DESCRIPTION
**Pull request by raoulstrackx**
Changes the standard library so that Nitro enclaves contact the enclave runner to set up outbound connections.

### CODE REVIEW

File: library/std/src/sys/unix/fortanixvme.rs [hunk 1]
```diff
@@ -0,0 +1,608 @@
...
+
+        pub fn socket(&self) -> &Socket {
+            &self.inner
+        }
+
+        pub fn into_socket(self) -> Socket {
+            self.inner
+        }
+
+        pub fn set_read_timeout(&self, dur: Option<Duration>) -> io::Result<()> {
```
Comment (Pagten) [inline library/std/src/sys/unix/fortanixvme.rs:276]: The unused `dur` parameter in this method and other unused parameters in other methods are causing compilation warnings.

File: library/std/src/sys/unix/fortanixvme.rs [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+            Ok(slot)
+        }
+    }
+
+    macro_rules! unimpl {
+        () => {
+            return Err(io::Error::new_const(
+                io::ErrorKind::Unsupported,
+                &"Unimplemented on Fortanixvme",
+            ));
```
Comment (Pagten) [inline library/std/src/sys/unix/fortanixvme.rs:140]: This trailing semicolon causes a warning on recent compiler versions.

File: library/std/src/sys/unix/fortanixvme.rs [hunk 3]
```diff
@@ -0,0 +1,608 @@
...
+            self.inner.is_read_vectored()
+        }
+
+        pub fn write(&self, buf: &[u8]) -> io::Result<usize> {
+            let len = cmp::min(buf.len(), <wrlen_t>::MAX as usize) as wrlen_t;
+            let ret = cvt(unsafe {
+                libc::send(self.inner.as_raw(), buf.as_ptr() as *const c_void, len, MSG_NOSIGNAL)
+            })?;
+            Ok(ret as usize)
+        }
```
Comment (Pagten) [inline library/std/src/sys/unix/fortanixvme.rs:315]: Why is this not simply `self.inner.write(buf)`?

File: library/std/src/sys/unix/fortanixvme.rs [hunk 4]
```diff
@@ -0,0 +1,608 @@
+pub mod net {
+    use core::convert::TryFrom;
+    use crate::cmp;
+    use crate::io::{self, ErrorKind, IoSlice, IoSliceMut};
+    use crate::sys::fd::FileDesc;
+    use crate::sys_common::{AsInner, FromInner, IntoInner};
```
Comment (Pagten) [inline library/std/src/sys/unix/fortanixvme.rs:6]: `AsInner` is unused

File: library/std/src/sys/unix/fortanixvme.rs [hunk 5]
```diff
@@ -0,0 +1,608 @@
...
+    use crate::time::Duration;
+    use crate::fmt;
+    use crate::mem;
+    use crate::net::{Ipv4Addr, Ipv6Addr, Shutdown, SocketAddr};
+    use crate::os::fd::raw::AsRawFd;
+    use crate::os::fd::owned::{AsFd, BorrowedFd};
+    use crate::os::unix::prelude::{IntoRawFd, FromRawFd, RawFd};
+    use crate::sys::{cvt, cvt_r};
```
Comment (Pagten) [inline library/std/src/sys/unix/fortanixvme.rs:14]: `cvt_r` is unused

File: library/std/src/sys/unix/fortanixvme.rs [hunk 6]
```diff
@@ -0,0 +1,608 @@
...
+    use fortanix_vme_abi::{self, Client, Error, Response, Request};
```
Comment (Pagten) [inline library/std/src/sys/unix/fortanixvme.rs:15]: `Response` and `Request` are unused

File: library/std/Cargo.toml [hunk 7]
```diff
@@ -41,6 +41,12 @@ dlmalloc = { version = "0.2.1", features = ['rustc-dep-of-std'] }
 [target.x86_64-fortanix-unknown-sgx.dependencies]
 fortanix-sgx-abi = { version = "0.3.2", features = ['rustc-dep-of-std'] }
 
+[target.x86_64-unknown-linux-fortanixvme.dependencies]
+fortanix-vme-abi = { path = "/home/raoul/Documents/rust-sgx/fortanix-vme/fortanix-vme-abi", default-features = false, features = ['rustc-dep-of-std'] }
```
Comment (raoulstrackx) [inline library/std/Cargo.toml:45]: Oops

Comment (raoulstrackx) [reply library/std/Cargo.toml:45]: Will update this when [rust-sgx PR345](https://github.com/fortanix/rust-sgx/pull/345) is merged

**Comment by Pagten** [comment]
bors d+

**Pull request merged**